contracts/contracts/Poll.sol

```diff
@@ -142,17 +155,29 @@ contract Poll is Params, Utilities, SnarkCommon, EmptyBallotRoots, IPoll {
       numMessages++;
     }
 
-    // init messageAq here by inserting placeholderLeaf
+    // init chainHash here by inserting placeholderLeaf
     uint256[2] memory dat;
     dat[0] = NOTHING_UP_MY_SLEEVE;
     dat[1] = 0;
 
     (Message memory _message, PubKey memory _padKey, uint256 placeholderLeaf) = padAndHashMessage(dat);
-    extContracts.messageAq.enqueue(placeholderLeaf);
+    chainHash = NOTHING_UP_MY_SLEEVE;
+    batchHashes.push(chainHash);
+    updateChainHash(placeholderLeaf);
 
     emit PublishMessage(_message, _padKey);
   }
 
+  /// @notice Returns reference to MACI contract
+  function getMaci() external view returns (IMACI) {
+    return maci;
+  }
+
+  // get all batch hash array elements
+  function getBatchHashes() external view returns (uint256[] memory) {
+    return batchHashes;
+  }
```

Comment:
Same here for batchHashes

Reply:
It is used in TS testing as fetching array of elements requires accessing per-index.